---
title: "cross_domain_network"
output:
  html_notebook:
    df_print: paged
    code_folding: hide
  editor_options: 
    chunk_output_type: inline
---
```{r include=FALSE}
library(qiime2R)
library(decontam)
library(ggeasy)
library(phyloseq)
library(Biostrings)
library(ggsignif)
library(ggeasy)
library(ggpubr)
library(readxl)
library(patchwork)
library(pairwiseAdonis)
library(DESeq2)
library(ALDEx2)
library(microbiome) 
library(vegan)
library(ANCOMBC)
library(metagMisc)
library(glue)
library(ggstatsplot)
library(beeswarm)
library(msa)
library(here)
library(ggtext)
library(kableExtra)
library(tidyHeatmap)
library(SpiecEasi)
library(NetCoMi)
library(igraph)
library(tidygraph)
library(ggraph)
library(tidyverse)
knitr::opts_chunk$set(fig.retina = 1, dpi=450)
options(dplyr.summarise.inform=F)
theme_set(theme_bw())
load(here('src/16_and_ITS_beta_diversity.RData'))
```
```{r setup-chunk, include=FALSE}
knitr::opts_chunk$set(dev = c('svg', 'png'),
                      dpi = 300,
                      echo = FALSE)

```
Created co-occurrence networks for each corn genotype to examine positive and negative correlations between taxa and look for differential associations between the two networks. In order to see what fungal or bacterial taxa are associated with Tox4 and K49, 16S and ITS abundances were combined into a cross domain network. To limit the size of the network, ASVs were limited to those that were present in at least 30% of the samples. This resulted in 84 bacterial ASVs and 15 fungal ASVs . Association matrices were created using SpiecEasi with the Meinshausen-Bühlmann neighborhood selection algorithm and then passed to NetCoMi’s netConstruct function to conduct a network comparison between the two groups.
```{r}
ps_phylum_filt_common_samples_pruned = ps_phylum_filt %>% map(~prune_samples(ps_phylum_filt_common_samples, .x))
ps_phylum_filt_common_samples_pruned_ra = map(ps_phylum_filt_common_samples_pruned, 
                                              ~transform_sample_counts(.x, function(x){x / sum(x)}))
ps_phylum_filt_common_samples_pruned_clr = map(ps_phylum_filt_common_samples_pruned, microbiome::transform, 'clr')
```
```{r}
ps_prev_high_b73 = map(ps_phylum_filt_common_samples_pruned, ~subset_samples(.x, Corn_Genotype == 'B73')) %>%
  map(~filter_taxa(.x, function(asv){(sum(asv > 0) > (nsamples(.x)*0.30))}, prune = T))
ps_prev_high_cml322 = map(ps_phylum_filt_common_samples_pruned, ~subset_samples(.x, Corn_Genotype == 'CML322')) %>%
  map(~filter_taxa(.x, function(asv){(sum(asv > 0) > (nsamples(.x)*0.30))}, prune = T) )

combined_taxa = map2(ps_prev_high_b73, ps_prev_high_b73, ~taxa_names(.x) %>% c() %>% unique())
lengths(combined_taxa)
ps_prev_high =  map2(ps_phylum_filt_common_samples_pruned, combined_taxa, ~prune_taxa(.y, .x))
ps_prev_high_ra = map(ps_prev_high, ~transform_sample_counts(.x, function(x){x / sum(x)}))
ps_prev_high_clr = map(ps_prev_high, microbiome::transform, 'clr')
cross_domain_tax = c(taxa_names(ps_prev_high$`16S`), taxa_names(ps_prev_high$ITS))
```

```{r, fig.width=12}
ps_prev_high_b73 = map(ps_prev_high, ~subset_samples(.x, Corn_Genotype == 'B73'))
ps_prev_high_cml322 = map(ps_prev_high, ~subset_samples(.x, Corn_Genotype == 'CML322'))
spiec_result_b73 = multi.spiec.easi(ps_prev_high_b73, 
                                 method='mb', nlambda=40, 
                                 lambda.min.ratio=1e-2, 
                                 pulsar.params = list(thresh = 0.05))
spiec_result_cml322 = multi.spiec.easi(ps_prev_high_cml322, 
                                 method='mb', nlambda=40, 
                                 lambda.min.ratio=1e-2, 
                                 pulsar.params = list(thresh = 0.05))
assoMat_b73 = SpiecEasi::symBeta(SpiecEasi::getOptBeta(spiec_result_b73), mode = "ave") %>% as.matrix()
assoMat_cml322 = SpiecEasi::symBeta(SpiecEasi::getOptBeta(spiec_result_cml322), mode = "ave") %>% as.matrix()
colnames(assoMat_b73) = rownames(assoMat_b73) = cross_domain_tax
diag(assoMat_b73) = 1
colnames(assoMat_cml322) = rownames(assoMat_cml322) = cross_domain_tax
diag(assoMat_cml322) = 1
net_b73_cml322 = netConstruct(data = assoMat_b73, data2 = assoMat_cml322, 
                              dataType = "condDependence", 
                              sparsMethod = "none", 
                              filtTax = "highestFreq",
                              filtTaxPar = list(highestFreq = 100))
netprops_b73_cml322 = netAnalyze(net_b73_cml322, hubPar = "eigenvector")
```


The plot shows the ASV co-abundance networks made with samples from each corn genotype. Light blue nodes represent bacterial ASVs and orange nodes represent fungal ASVs. The direction of the association is shown by the color of the edges with red representing negative edges and green representing positive edges. The ASVs are labelled with their genotype truncated to 10 characters to minimize overlap. The size of the nodes are scaled according to the node's centrality. Hub nodes are indicated by bold text and node border.

```{r, fig.width=13, fig.height=6}
nodeCols = c(rep("lightblue", ntaxa(ps_prev_high$`16S`)), rep("orange", ntaxa(ps_prev_high$ITS)))
names(nodeCols) = cross_domain_tax
labels  = c(get_tax(ps_prev_high$`16S`)$Genus,
                     get_tax(ps_prev_high$ITS)$Genus) %>%
  substr(1, 10)
labels = bind_rows(get_tax(ps_prev_high$`16S`), get_tax(ps_prev_high$ITS)) %>%
  mutate(label = coalesce(Genus, Order) %>% substr(1, 10)) %>%
    select(feature_id, label) %>% deframe()
names(labels) = cross_domain_tax
labels = labels %>% enframe() %>%
  mutate(value = case_when(name == 'f238a589635f01ddc3a300a2f6e63322' ~ 'A_flavus_Tox4', 
                           name == '06ee4aa94b2b7521c7b13c1fb07879bf' ~ 'A_flavus_K49',
                           TRUE ~ value)) %>%
  deframe()
plot(netprops_b73_cml322, 
     labels = labels,
     sameLayout = TRUE, 
     layoutGroup = "union",
     nodeColor = "colorVec", 
     colorVec = nodeCols,
     nodeSize = "eigen", 
     nodeSizeSpread = 1.5, 
     labelScale = FALSE,
     cexNodes = 2, 
     cexLabels = 0.8,
     cexHubLabels = 0.8,
     cexTitle = 2,
     groupNames = c("B73", "CML322"),
     rmSingles = 'inboth',
     nodeFilter = 'highestConnect',
     nodeFilterPar = 60,
     repulsion=0.99)
legend(-0.2, 1.2, cex = 1.3, pt.cex = 1.3, 
       legend = c("Bacteria", "Fungi"), col = c("lightblue", "orange"), 
       bty = "n", pch = 16) 
```

```{r, echo=FALSE}
netcomp_b73_cml322 = netCompare(netprops_b73_cml322, permTest = FALSE) 
net_comp_summary = summary(netcomp_b73_cml322, groupNames = c("B73", "CML322"), digits=2L)
```

<br><br><br>
Number of bacterial and fungal ASVs in each network:
```{r}
net_b73_cml322$edgelist1 %>% mutate(corn_genotype = 'B73') %>%
  bind_rows(net_b73_cml322$edgelist2 %>% mutate(corn_genotype = 'CML322')) %>% 
    pivot_longer(cols=c(v1, v2), names_to='from_to', values_to='feature_id') %>%
    distinct(corn_genotype, feature_id) %>%
    group_by(corn_genotype) %>%
    summarize(num_fungal_nodes = sum(feature_id %in% get_counts(ps_phylum_filt_common_samples_pruned$ITS)$feature_id),
              num_bacterial_nodes = sum(feature_id %in%
                                          get_counts(ps_phylum_filt_common_samples_pruned$`16S`)$feature_id))
```
Number of bacterial and fungal ASVs in each network:
```{r}
net_b73_cml322$edgelist1 %>% mutate(corn_genotype = 'B73') %>%
  bind_rows(net_b73_cml322$edgelist2 %>% mutate(corn_genotype = 'CML322')) %>% 
    pivot_longer(cols=c(v1, v2), names_to='from_to', values_to='feature_id') %>%
    distinct(feature_id) %>%
    mutate(kingdom = if_else(feature_id %in% get_counts(ps_phylum_filt_common_samples_pruned$ITS)$feature_id,
                                      'fungi', 'bacteria')) %>%
    count(kingdom)
```

Hub taxa in each network
```{r}
tibble(feature_id = netprops_b73_cml322$hubs$hubs1, network = 'B73') %>%
  bind_rows(tibble(feature_id = netprops_b73_cml322$hubs$hubs2, network = 'CML322')) %>%
  left_join(map(ps_phylum_filt_common_samples_pruned, get_tax) %>% bind_rows(), by='feature_id') %>%
  select(-c(feature_id, Kingdom:Family))
```

<br><br><br>
This shows the overall distribution of edge weights in the network, and puts edges that have A. flavus Tox4 and K49 in the context of the rest of the network. The edges containing A. flavus are not very strong.
```{r, fig.height=3}
get_flavus_edges = . %>%
  filter(v1 %in% c('f238a589635f01ddc3a300a2f6e63322', '06ee4aa94b2b7521c7b13c1fb07879bf') |
           v2 %in% c('f238a589635f01ddc3a300a2f6e63322', '06ee4aa94b2b7521c7b13c1fb07879bf')) %>%
    rowwise() %>%
  mutate(flavus_ASV = intersect(c('f238a589635f01ddc3a300a2f6e63322', '06ee4aa94b2b7521c7b13c1fb07879bf'),
                                c(v1, v2))) %>%
  mutate(other_ASV = setdiff(c(v1, v2),
                              c('f238a589635f01ddc3a300a2f6e63322', '06ee4aa94b2b7521c7b13c1fb07879bf'))) %>%
  select(-c(v1, v2, adja, diss)) %>%
  
  left_join(bind_rows(ps_phylum_filt_tax), by=c('other_ASV' = 'feature_id')) %>%
  arrange(desc(asso))
flavus_edges_b73 = net_b73_cml322$edgelist1 %>% get_flavus_edges()
flavus_edges_cml322 = net_b73_cml322$edgelist2 %>% get_flavus_edges()
flavus_edges_b73_cml322 = flavus_edges_b73 %>% mutate(corn_genotype = 'B73') %>%
  bind_rows(flavus_edges_cml322 %>% mutate(corn_genotype = 'CML322')) %>%
  relocate(corn_genotype) 
flavus_edges_b73_cml322 %>% 
  mutate(flavus_ASV = str_replace_all(flavus_ASV, c('06ee4aa94b2b7521c7b13c1fb07879bf'= 'K49_ASV', 
                           'f238a589635f01ddc3a300a2f6e63322' ='Tox4_ASV'))) %>%
  select(corn_genotype, flavus_ASV, weight=asso, Order, Genus, Species) 
net_b73_cml322$edgelist1 %>%
  bind_rows(net_b73_cml322$edgelist1) %>%
  mutate(flavus_edge = v1 %in% c('f238a589635f01ddc3a300a2f6e63322', '06ee4aa94b2b7521c7b13c1fb07879bf') |
           v2 %in% c('f238a589635f01ddc3a300a2f6e63322', '06ee4aa94b2b7521c7b13c1fb07879bf')) %>%
  rename(weight = asso) %>%
  ggplot(aes(x=weight, fill=flavus_edge)) +
  geom_histogram(bins=20) +
  labs(title='Histogram of edge weights') +
  ggeasy::easy_center_title()
```
<br><br><br>
Plotted just the A. flavus ASVs and the ASVs coabundant with them 
```{r, fig.width=13, fig.height=6}
plot(netprops_b73_cml322, 
     labels = labels,
     sameLayout = TRUE, 
     layoutGroup = "union",
     nodeColor = "colorVec", 
     colorVec = nodeCols,
     nodeSize = "eigen", 
     nodeSizeSpread = 1.5, #2,
     labelScale = FALSE,
     cexNodes = 2, 
     cexLabels = 0.8,
     cexHubLabels = 0.8,
     cexTitle = 2,
     groupNames = c("B73", "CML322"),
     rmSingles = 'inboth',
     nodeFilter = 'names',
     nodeFilterPar = c(flavus_edges_b73$flavus_ASV, flavus_edges_b73$other_ASV, 
                       flavus_edges_cml322$flavus_ASV, flavus_edges_cml322$other_ASV) %>% unique(),
     repulsion=0.99)
#legend(-0.2, 1.2, cex = 1.3, pt.cex = 1.3, 
#       legend = c("Bacteria", "Fungi"), col = c("lightblue", "orange"), 
#       bty = "n", pch = 16) 
```


```{r, fig.width=10}
#What does the relative abundances of the highest edges look like? 
ps_prev_high_ra %>% map(get_counts) %>% bind_rows() %>% 
  pivot_longer(-feature_id, names_to = 'SampleID', values_to = 'relative_counts') %>%
  right_join(
    net_b73_cml322$edgelist1 %>%
      arrange(desc(asso)) %>%
      mutate(pair_num = row_number()) %>%
      slice_max(asso, n=5) %>% 
      pivot_longer(cols=c(v1, v2), names_to='from_to', values_to='feature_id'),
      by='feature_id'
) %>% 
  mutate(feature_id = if_else(from_to == 'v1', 'v1', 'v2')) %>%
  inner_join(ps_prev_high_ra$ITS %>% get_sample_data() %>% filter(Corn_Genotype == 'B73'), by='SampleID') %>%
  ggplot() +
    geom_line(aes(condition_w_rep, relative_counts, group=feature_id, color=feature_id)) +
    facet_wrap(feature_id ~ pair_num, scales = 'free_y', nrow=2)  +
    theme(axis.text.x=element_text(angle=90,hjust=1)) +
    labs(title=str_glue('Relative abundance of ASV pairs with highest edge weights'),
         y='relative abundance') + 
    ggeasy::easy_remove_legend() 

```

Tested for differential associations between the B73 and CML322 networks and did not find any significant differential associations.
```{r}
diffnet_b73_cml322 = diffnet(net_b73_cml322, diffMethod = 'fisherTest', 
        n1=nsamples(ps_prev_high_b73$ITS), n2=nsamples(ps_prev_high_cml322$ITS))
```

